src/lib/server/preprocessMessages.ts

```diff
@@ -13,11 +13,9 @@ export async function preprocessMessages(
 	return await Promise.all(
 		structuredClone(messages).map(async (message, idx) => {
 			const webSearchContext = webSearch?.contextSources
-				.map(({ context }) => context)
-				.flat()
-				.sort((a, b) => a.idx - b.idx)
-				.map(({ text }) => text)
-				.join(" ");
+				.map(({ context }) => context.trim())
+				.join("\n\n----------\n\n");
```

Comment:
why is `.join("\n\n----------\n\n")` used over `.join(" ");` ?

Reply:
The content is formatted in markdown so I thought it might be helpful for the model to know when it's receiving a different source but not sure if it's helping that much